examples/helpers/tiled.rs

```diff
@@ -135,115 +181,159 @@ pub fn process_loaded_maps(
                     // commands.entity(*layer_entity).despawn_recursive();
                 }
 
-                for tileset in tiled_map.map.tilesets.iter() {
-                    // Once materials have been created/added we need to then create the layers.
-                    for layer in tiled_map.map.layers.iter() {
-                        let tile_size = TilemapTileSize {
-                            x: tileset.tile_width as f32,
-                            y: tileset.tile_height as f32,
-                        };
-                        let tile_spacing = TilemapSpacing {
-                            x: tileset.spacing as f32,
-                            y: tileset.spacing as f32,
-                        };
+                // The TilemapBundle requires that all tile images come exclusively from a single
+                // tiled texture or from a Vec of independent per-tile images. Furthermore, all of
+                // the per-tile images must be the same size. Since Tiled allows tiles of mixed
+                // tilesets on each layer and allows differently-sized tile images in each tileset,
+                // this means we need to load each combination of tileset and layer separately.
+                for (tileset_index, tileset) in tiled_map.map.tilesets().iter().enumerate() {
+                    let tile_size = TilemapTileSize {
+                        x: tileset.tile_width as f32,
+                        y: tileset.tile_height as f32,
+                    };
+
+                    let tile_spacing = TilemapSpacing {
+                        x: tileset.spacing as f32,
+                        y: tileset.spacing as f32,
+                    };
 
+                    // Once materials have been created/added we need to then create the layers.
+                    for (layer_index, layer) in tiled_map.map.layers().enumerate() {
                         let offset_x = layer.offset_x;
                         let offset_y = layer.offset_y;
 
-                        let map_size = TilemapSize {
-                            x: tiled_map.map.width,
-                            y: tiled_map.map.height,
+                        let tiled::LayerType::TileLayer(tile_layer) = layer.layer_type() else {
+                            log::info!(
+                                "Skipping layer because {:?} is not supported.",
```

Comment:
It seems like the debug impl for `LayerType` prints **all** the map data somehow, making this spew several screens of spam for my map.

Maybe something like
```rust
log::info!(
    "Skipping layer {} because it is an unsupported type",
    layer.id()
);
```
would be adequate?